essence-of-live-coding/test/Cell/Util.hs

```diff
@@ -150,3 +164,58 @@ test = testGroup "Utility unit tests"
     , output = [[0,0,0],[1,1],[2,2,0]]
     }
   ]
+
+testTraverse' :: Test
+testTraverse' = testGroup "Traversing unit tests"
+  [ testProperty "traverse' (arr f) = arr (f <$>) ([a])"
+    $ \(input :: [[Int]], Fn (f :: Int -> Int)) -> traverseArrLaw f input
+  , testProperty "traverse' (arr f) = arr (f <$>) (Maybe a)"
+    $ \(input :: [Maybe Int], Fn (f :: Int -> Int)) -> traverseArrLaw f input
+  , testProperty "traverse' (arr f) = arr (f <$>) (Vecttor 4 Int)"
+    $ \(input :: [(Int,Int,Int,Int)], Fn (f :: Int -> Int)) -> traverseArrLaw f (V.fromTuple <$> input)
+  , testProperty "traverse' works as expected for any Cell Identy Int Int created with constructor Cell ([a])"
+    $ \(input :: [[Int]], s :: Int, Fn (f :: (Int, Int) -> (Int,Int))) -> traverseCellTest s (curry f) input
+  , testProperty "traverse' works as expected for any Cell Identy Int Int created with constructor Cell (Maybe a)"
+    $ \(input :: [Maybe Int], s :: Int, Fn (f :: (Int, Int) -> (Int,Int))) -> traverseCellTest s (curry f) input
+  , testProperty "traverse' works as expected for any Cell Identy Int Int created with constructor Cell (Vecttor 4 Int)"
+    $ \(input :: [(Int,Int,Int,Int)], s :: Int, Fn (f :: (Int, Int) -> (Int,Int))) -> traverseCellTest s (curry f) (V.fromTuple <$> input)
+  , testCase "traverse' by itself does not force the entire list (ArrM)"
+    $ testError "traverse' is too strict"
+    $ head $ head $ fst $ runIdentity $ steps (traverse' (arr id) :: Cell Identity [Int] [Int]) [1 : error "Bang !"]
+  , testCase "traverse' by itself does not force the entire list (Cell)"
+    $ testError "Traverse' is too strict"
+    $ head $ head $ fst $ runIdentity $ steps (traverse' cellId :: Cell Identity [Int] [Int]) [1 : error "Bang !"]
+  , testCase "traverse' of a Cell composed with another Cell does not force the entire list"
+    $ testError "Traverse' is too strict"
```

Comment:
Ok, so this tests that in fact this throws an exception, or not?

Reply:
The test succeeds if it does not throw an exception. So in this case as you mentioned elsewhere it is simpler to just test with CellSimulation, since if it is too strict then it will throw an exception and that will consider the test as failed.